File: javascript/bracket-push/bracket-push.js
function closingBracket(br) {
	if (('}' === br) || (']' === br) || (')' === br)) {
		return true;
	}
	
	return false;
}

function bracketMatch(opening, closing) {
	if (('(' === opening) && (')' === closing)) {
		return true;
	}
	
	if (('[' === opening) && (']' === closing)) {
		return true;
	}
	
	if (('{' === opening) && ('}' === closing)) {
		return true;
	}
	
	return false;
}

Array.prototype.empty = function() {
	if(this.length <= 0) {
		return true;
	}
	
	return false;
};

function bracket(input) {
	var stack = [];
	
	for (var i = 0; i < input.length; ++i) {
		if (closingBracket(input[i])) {
			var poppedMatch = stack.pop();

			if (!bracketMatch(poppedMatch, input[i])) {
				return false;
			}
		} else {
			stack.push(input[i]);
		}
	}

	return stack.empty();
}

module.exports = bracket;

File: python/bracket-push/bracket_push.py
def closing_bracket(br):
    if ('}' == br) or (']' == br) or (')' == br):
        return True
    
    return False

def bracket_match(opening, closing):
    if ('(' == opening) and (')' == closing):
        return True
    
    if ('[' == opening) and (']' == closing):
        return True
    
    if ('{' == opening) and ('}' == closing):
        return True
    
    return False

def pop_noexcept(stack):
    return stack.pop() if not (len(stack) <= 0) else -1

def check_brackets(br_input):
    stack = []

    for bracket in br_input:
        if closing_bracket(bracket):
            if not bracket_match(pop_noexcept(stack), bracket):
                return False
        else:
            stack.append(bracket)

    return (len(stack) <= 0)